packages/perseus/src/components/math-input.tsx

```diff
@@ -75,6 +75,11 @@ type State = {
     keypadOpen: boolean;
     cursorContext: typeof CursorContext[keyof typeof CursorContext];
     openedWithEventType?: string;
+    // [count, label]
+    // count is just used to trigger a re-render when the label changes,
+    // even if the label is the same as the previous value.
+    // label is the last character typed.
+    lastChar: Array<number | string>;
```

Comment:
Why do we want to be able to explicitly trigger a re-render? If the intention is to get screenreaders to announce aria-live contents, I'm not sure this will always work. I think React will diff the new label string with its memory of the previous render, see that nothing changed, and skip updating the DOM.

If you're finding that this does actually work with a screenreader, I'd like to understand why it works.

If we decide to keep both the count and the label, could they both be top-level state properties? I.e.

```suggestion
    // count is just used to trigger a re-render
    count: number;
    // used in a role="alert" element to provide feedback as the user types
    lastCharacterTyped: string;
```

I feel like that's simpler, and more likely to be what future readers of this code will expect to see. (It's closer to what I expected, anyway).

Reply:
Ah, you're absolutely right. I was taking inspiration from this [screen reader alert component](https://github.com/Khan/webapp/blob/42a81b6872bc37c002af67107b903fbdd60bad07/services/static/javascript/python-coding/screen-reader-alert.tsx), but I totally missed the `key` prop. I'll update it now!